src/Promenade.Web/Views/Home/Locations.cshtml

```diff
@@ -0,0 +1,160 @@
+                                                        </td>
+                                                    </tr>
+                                                    <tr>
+                                                        <td style="padding-bottom:10px;">
+                                                            <strong class="traffic-green">Open until 5:00 PM</strong>
+                                                        </td>
+                                                    </tr>
+                                                    <tr style="text-align: left;">
+                                                        <td style="padding: 5px 0;">
+                                                            <a href="Locations/@Model.CloseLocations[i].Stub" target="_blank" class="btn btn-default">Library Info</a>
+                                                            <a href="@Model.CloseLocations[i].EventLinkId" class="btn btn-default" style="float:right" target="_blank">Events</a>
+                                                        </td>
+
+                                                    </tr>
+                                                </tbody>
+                                            </table>
+                                        </div>
+                                    </div>
+                                </div>
+                            </div>
+                        </text>
+                    }
+                }
+                else
+                {
+                    <script>
+                        var noResults = true;
+                    </script>
+                }
+            </div>
+        </div>
+        @section scripts {
+            <script>
+        var useMyLocation = 'Use my location';
+
+        $().ready(function () {
+            if (location.protocol != 'https:' && location.hostname != 'localhost') {
+                location.href = 'https:' + window.location.href.substring(window.location.protocol.length);
+            }
+            if (navigator.geolocation) {
+                $('#GeoButton').html(useMyLocation);
+                $('#GeoButton').show();
+            }
+        });
+
+        $('#LocatorAddress').ready(function () {
+            if (typeof noResults !== 'undefined') {
+                $('#LocatorAddress').focus();
+            }
+        });
+
+        var geoSuccess = function (position) {
+            $('#Latitude').val(position.coords.latitude);
+            $('#Longitude').val(position.coords.longitude);
+            $('#GeoForm').submit();
+        }
+
+        var geoFailure = function (error) {
+            $('#GeoButton').html(useMyLocation);
+            $('#GeoButton').prop('disabled', false);
+            alert('There was a problem determining your location, please enter an address or zip code to find your nearest library.');
```

Comment:
Can't have "Library" here, needs to come from the ViewModel or database somehow.